Trilinos nightly build failures: error: more than one instance of overloaded function "Kokkos::Impl::create_mirror" matches the argument list

The nightly Trilinos serial build with intel/18 began failing with `error: more than one instance of overloaded function "Kokkos::Impl::create_mirror" matches the argument list` message, indicating likely related to #4486 

**Error message**:
```
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/kokkos/core/src/Kokkos_CopyViews.hpp(3316): error: more than one instance of overloaded function "Kokkos::Impl::create_mirror" matches the argument list:
19:23:21             function template "Kokkos::View<T, P...>::HostMirror Kokkos::create_mirror(const Kokkos::View<T, P...> &, std::enable_if<<expression>, void>::type *)"
19:23:21             function template "Kokkos::View<T, P...>::HostMirror Kokkos::create_mirror(const Kokkos::View<T, P...> &)"
19:23:21             argument types are: (const Kokkos::View<const panzer::Traits::Tangent::ScalarT ***, Kokkos::Serial::array_layout, Kokkos::DefaultExecutionSpace>)
19:23:21     return create_mirror(src, arg_prop...);
19:23:21            ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/kokkos/core/src/Kokkos_CopyViews.hpp(3287): note: this candidate was rejected because mismatch in count of arguments
19:23:21   create_mirror(const Space&, const Kokkos::View<T, P...>& src,
19:23:21   ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/kokkos/core/src/Kokkos_CopyViews.hpp(3253): note: this candidate was rejected because at least one template argument could not be deduced
19:23:21   create_mirror(const Kokkos::View<T, P...>& src, const I&... arg_prop) {
19:23:21   ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/kokkos/core/src/Kokkos_CopyViews.hpp(3230): note: this candidate was rejected because at least one template argument could not be deduced
19:23:21   create_mirror(const Kokkos::View<T, P...>& src, const I&... arg_prop) {
19:23:21   ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/kokkos/containers/src/Kokkos_DynRankView.hpp(1974): note: this candidate was rejected because mismatch in count of arguments
19:23:21   typename Impl::MirrorDRVType<Space, T, P...>::view_type create_mirror(
19:23:21                                                           ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/kokkos/containers/src/Kokkos_DynRankView.hpp(1959): note: this candidate was rejected because at least one template argument could not be deduced
19:23:21   inline typename DynRankView<T, P...>::HostMirror create_mirror(
19:23:21                                                    ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/kokkos/containers/src/Kokkos_DynRankView.hpp(1945): note: this candidate was rejected because at least one template argument could not be deduced
19:23:21   inline typename DynRankView<T, P...>::HostMirror create_mirror(
19:23:21                                                    ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/packages/sacado/src/Kokkos_DynRankView_Fad.hpp(1246): note: this candidate was rejected because mismatch in count of arguments
19:23:21   create_mirror(const Space& , const Kokkos::DynRankView<T,P...> & src
19:23:21   ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/packages/sacado/src/Kokkos_DynRankView_Fad.hpp(1198): note: this candidate was rejected because at least one template argument could not be deduced
19:23:21   create_mirror( const Kokkos::DynRankView<T,P...> & src
19:23:21   ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/packages/sacado/src/Kokkos_DynRankView_Fad.hpp(1173): note: this candidate was rejected because at least one template argument could not be deduced
19:23:21   create_mirror( const Kokkos::DynRankView<T,P...> & src
19:23:21   ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/kokkos/core/src/Kokkos_CopyViews.hpp(3362): note: this candidate was rejected because mismatch in count of arguments
19:23:21   typename Impl::MirrorType<Space, T, P...>::view_type create_mirror(
19:23:21                                                        ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/kokkos/core/src/Kokkos_CopyViews.hpp(3355): note: this candidate was rejected because mismatch in count of arguments
19:23:21   typename Impl::MirrorType<Space, T, P...>::view_type create_mirror(
19:23:21                                                        ^
19:23:21 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/kokkos/core/src/Kokkos_CopyViews.hpp(3348): note: this candidate was rejected because mismatch in count of arguments
19:23:21   typename Kokkos::View<T, P...>::HostMirror create_mirror(
19:23:21                                              ^
19:23:21           detected during:
19:23:21             instantiation of "std::enable_if<<expression>, Kokkos::View<T, P...>::HostMirror>::type Kokkos::Impl::create_mirror_view(const Kokkos::View<T, P...> &, const I &...) [with T=const panzer::Traits::Tangent::ScalarT ***, P=<Kokkos::Serial::array_layout, Kokkos::DefaultExecutionSpace>, I=<>]" at line 3371
19:23:21             instantiation of "Kokkos::View<T, P...>::HostMirror Kokkos::create_mirror_view(const Kokkos::View<T, P...> &) [with T=const panzer::Traits::Tangent::ScalarT ***, P=<Kokkos::Serial::array_layout, Kokkos::DefaultExecutionSpace>]" at line 164 of "/home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/packages/panzer/disc-fe/src/evaluators/Panzer_Dirichlet_Residual_EdgeBasis_impl.hpp"
19:23:21             instantiation of "void panzer::DirichletResidual_EdgeBasis<EvalT, Traits>::evaluateFields(Traits::EvalData) [with EvalT=panzer::Traits::Tangent, Traits=panzer::Traits]" at line 52 of "/home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/packages/panzer/disc-fe/src/evaluators/Panzer_Dirichlet_Residual_EdgeBasis.cpp"
19:23:21 
19:23:22 /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/packages/panzer/disc-fe/src/evaluators/Panzer_Dirichlet_Residual_EdgeBasis_impl.hpp(100): warning #1478: function "PHX::EvaluatorWithBaseImpl<Traits>::addDependentField(const PHX::MDField<DataT, Props...> &) [with Traits=panzer::Traits, DataT=double, Props=<panzer::Cell, panzer::IP, panzer::Dim, panzer::Dim>]" (declared at line 108 of
19:23:22           "/home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/packages/phalanx/src/Phalanx_Evaluator_WithBaseImpl.hpp") was declared deprecated
19:23:22     this->addDependentField(pointValues.jac);
19:23:22           ^
19:23:22           detected during instantiation of "panzer::DirichletResidual_EdgeBasis<EvalT, Traits>::DirichletResidual_EdgeBasis(const Teuchos::ParameterList &) [with EvalT=panzer::Traits::Jacobian, Traits=panzer::Traits]" at line 52 of "/home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/packages/panzer/disc-fe/src/evaluators/Panzer_Dirichlet_Residual_EdgeBasis.cpp"
19:23:22 
19:23:22 compilation aborted for /home/jenkins/blake-new/workspace/KokkosEco_Trilinos_Blake_Intel181163_Serial_Cplx/Trilinos/packages/panzer/disc-fe/src/evaluators/Panzer_Dirichlet_Residual_EdgeBasis.cpp (code 2)
19:23:22 make[2]: *** [packages/panzer/disc-fe/src/CMakeFiles/panzer-disc-fe.dir/build.make:1053: packages/panzer/disc-fe/src/CMakeFiles/panzer-disc-fe.dir/evaluators/Panzer_Dirichlet_Residual_EdgeBasis.cpp.o] Error 2
```

A quick glance at the marked `create_mirror_view` line from the compiler at https://github.com/trilinos/Trilinos/blob/37c24f45d070d06702906186e0e8ef908a6c59b9/packages/panzer/disc-fe/src/evaluators/Panzer_Dirichlet_Residual_EdgeBasis_impl.hpp#L164 doesn't indicate any obvious funny business...


**Reproducer**:
```
# Clone repos
git clone -b kokkos-promotion git@github.com:trilinos/Trilinos.git
git clone -b develop git@github.com:kokkos/kokkos.git

# Create symbolic link to kokkos within Trilinos source directory for source override
cd Trilinos
ln -s ../kokkos kokkos
cd -

# Create build directory
mkdir -p Build
cd Build

# Load env
export TRILINOS_DIR=$PWD/../Trilinos
export ATDM_CONFIG_REGISTER_CUSTOM_CONFIG_DIR=${TRILINOS_DIR}/cmake/std/atdm/contributed/blake
source ${TRILINOS_DIR}/cmake/std/atdm/load-env.sh intel-serial

export ARCH_CXX_FLAG="-xCORE-AVX512 -mkl"
export ARCH_C_FLAG="-xCORE-AVX512 -mkl"
export BLAS_LIBRARIES="-mkl;${MKLROOT}/lib/intel64/libmkl_intel_lp64.a;${MKLROOT}/lib/intel64/libmkl_intel_thread.a;${MKLROOT}/lib/intel64/libmkl_core.a"
export LAPACK_LIBRARIES=${BLAS_LIBRARIES}


# Top Level Configuration Options
TESTS=ON
EXAMPLES=ON
SHARED=ON

CUDA=OFF
OPENMP=OFF
PTHREAD=OFF
SERIAL=ON
COMPLEX=ON
ENABLE_SHYLU="ON"

# Configure
cmake \
-GNinja \
-DCMAKE_INSTALL_PREFIX="${TRILINOS_INSTALL_DIR}" \
-DCMAKE_CXX_STANDARD="14" \
-D Trilinos_ENABLE_COMPLEX_DOUBLE=${COMPLEX} \
\
-D Kokkos_ARCH_SKX=ON \
-D CMAKE_CXX_FLAGS="-g" \
-D CMAKE_C_FLAGS="${ARCH_C_FLAG} -g" \
-D CMAKE_Fortran_FLAGS="${ARCH_C_FLAG} -g" \
-D CMAKE_EXE_LINKER_FLAGS="${ARCH_C_FLAG}" \
-D CMAKE_Fortran_COMPILER="mpif77" \
-D HAVE_CXX_PRAGMA_WEAK:BOOL=OFF \
-D CMAKE_AR:FILEPATH=/usr/bin/ar \
-D CMAKE_STRIP:FILEPATH=/usr/bin/strip \
-D CMAKE_RANLIB:FILEPATH=/usr/bin/ranlib \
-D Trilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON \
-D Trilinos_ENABLE_DEBUG:BOOL=OFF \
\
-D Trilinos_ENABLE_INSTALL_CMAKE_CONFIG_FILES:BOOL=ON \
-D CMAKE_BUILD_TYPE:STRING=RELEASE \
-D CMAKE_VERBOSE_MAKEFILE:BOOL=OFF \
-D CMAKE_SKIP_RULE_DEPENDENCY=ON \
-D Trilinos_ENABLE_CHECKED_STL:BOOL=OFF \
-D Trilinos_ENABLE_ALL_PACKAGES:BOOL=OFF \
-D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF \
-D BUILD_SHARED_LIBS:BOOL=${SHARED} \
-D DART_TESTING_TIMEOUT:STRING=500 \
-D Trilinos_WARNINGS_AS_ERRORS_FLAGS:STRING="" \
\
-D Trilinos_ENABLE_OpenMP=${OPENMP} \
-D TPL_ENABLE_CUDA=${CUDA} \
-D TPL_ENABLE_MPI=ON \
  -D MPI_EXEC_POST_NUMPROCS_FLAGS:STRING="-bind-to;socket;-map-by;socket" \
-D TPL_ENABLE_BinUtils=OFF \
-D TPL_ENABLE_SuperLU=OFF \
  -D TPL_SuperLU_LIBRARIES:STRING="${SUPERLU_ROOT}/lib/libsuperlu.a" \
  -D TPL_SuperLU_INCLUDE_DIRS:STRING="${SUPERLU_ROOT}/include" \
-D TPL_ENABLE_BLAS=ON \
  -D TPL_BLAS_LIBRARIES:PATH="${BLAS_LIBRARIES}" \
-D TPL_ENABLE_LAPACK=ON \
  -D TPL_LAPACK_LIBRARIES:PATH="${LAPACK_LIBRARIES}" \
-D TPL_ENABLE_Boost=ON \
   -D Boost_INCLUDE_DIRS:PATH="${BOOST_ROOT}/include" \
   -D Boost_LIBRARY_DIRS:PATH="${BOOST_ROOT}/lib" \
-D TPL_ENABLE_BoostLib=ON \
   -D BoostLib_INCLUDE_DIRS:PATH="${BOOST_ROOT}/include" \
   -D BoostLib_LIBRARY_DIRS:PATH="${BOOST_ROOT}/lib" \
-D TPL_ENABLE_Netcdf=ON \
  -D Netcdf_INCLUDE_DIRS:PATH="${NETCDF_ROOT}/include" \
  -D Netcdf_LIBRARY_DIRS:PATH="${NETCDF_ROOT}/lib" \
  -D TPL_Netcdf_LIBRARIES:PATH="${NETCDF_ROOT}/lib/libnetcdf.a;${HDF5_ROOT}/lib/libhdf5_hl.a;${HDF5_ROOT}/lib/libhdf5.a;${ZLIB_ROOT}/lib/libz.a;${PNETCDF_ROOT}/lib/libpnetcdf.a" \
  -D TPL_Netcdf_PARALLEL:BOOL=ON \
-D TPL_ENABLE_HDF5=ON \
  -D HDF5_INCLUDE_DIRS:PATH="${HDF5_ROOT}/include" \
  -D TPL_HDF5_LIBRARIES:PATH="${HDF5_ROOT}/lib/libhdf5_hl.a;${HDF5_ROOT}/lib/libhdf5.a;${ZLIB_ROOT}/lib/libz.a" \
-D TPL_ENABLE_Zlib=ON \
  -D Zlib_INCLUDE_DIRS:PATH="${ZLIB_ROOT}/include" \
  -D TPL_Zlib_LIBRARIES:PATH="${ZLIB_ROOT}/lib/libz.a" \
-D TPL_ENABLE_DLlib=ON \
\
\
-D Trilinos_ENABLE_Amesos=ON \
-D Trilinos_ENABLE_Amesos2=ON \
  -D Amesos2_ENABLE_TESTS:BOOL=${TESTS} \
  -D Amesos2_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
-D Trilinos_ENABLE_ShyLU_NodeTacho=${ENABLE_SHYLU} \
  -D ShyLU_NodeTacho_ENABLE_TESTS:BOOL=${TESTS} \
  -D ShyLU_NodeTacho_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
-D Trilinos_ENABLE_Anasazi=ON \
  -D Anasazi_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_AztecOO=ON \
-D Trilinos_ENABLE_Belos=ON \
  -D Belos_ENABLE_TESTS:BOOL=${TESTS} \
  -D Belos_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
-D Trilinos_ENABLE_Epetra=ON \
-D Trilinos_ENABLE_EpetraExt=ON \
-D Trilinos_ENABLE_Ifpack=ON \
  -D Ifpack_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_Ifpack2=ON \
  -D Ifpack2_ENABLE_TESTS:BOOL=${TESTS} \
  -D Ifpack2_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
-D Trilinos_ENABLE_Intrepid=ON \
  -D Intrepid_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
  -D Intrepid_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_Intrepid2=ON \
  -D Intrepid2_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
  -D Intrepid2_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_Isorropia=ON \
  -D Isorropia_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_Kokkos=ON \
  -D Kokkos_ENABLE_SERIAL=${SERIAL} \
  -D Kokkos_ENABLE_PTHREAD=${PTHREAD} \
  -D Kokkos_ENABLE_OPENMP=${OPENMP} \
  -D Kokkos_ENABLE_CUDA=${CUDA} \
  -D Kokkos_ENABLE_CUDA_LAMBDA=${CUDA} \
  -D Kokkos_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_KokkosKernels=ON \
  -D KokkosKernels_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_ML=ON \
  -D ML_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_Moertel=OFF \
  -D Moertel_ENABLE_EXAMPLES=OFF \
-D Trilinos_ENABLE_MueLu=ON \
  -D MueLu_INST_DOUBLE_INT_LONGINT:BOOL=ON \
  -D MueLu_ENABLE_TESTS:BOOL=${TESTS} \
  -D MueLu_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
  -D MueLu_ENABLE_Experimental:BOOL=ON \
  -D MueLu_ENABLE_Kokkos_Refactor:BOOL=ON \
  -D MueLu_ENABLE_Epetra:BOOL=OFF \
  -D Trilinos_ENABLE_NOX=ON \
  -D NOX_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_Panzer=ON \
  -D Panzer_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_Phalanx=ON \
  -D Phalanx_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_ROL=ON \
  -D ROL_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_Sacado=ON \
  -D Sacado_ENABLE_UNINIT:BOOL=ON \
  -D Sacado_ENABLE_TESTS:BOOL=${TESTS} \
  -D Sacado_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
-D Trilinos_ENABLE_Stokhos=ON \
  -D Stokhos_ENABLE_TESTS:BOOL=${TESTS} \
  -D Stokhos_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
-D Trilinos_ENABLE_Stratimikos=ON \
  -D Stratimikos_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_Thyra=ON \
  -D Thyra_ENABLE_TESTS:BOOL=${TESTS} \
-D Trilinos_ENABLE_Tpetra=ON \
  -D Tpetra_INST_SERIAL:BOOL=${SERIAL} \
  -D Tpetra_INST_OPENMP:BOOL=${OPENMP} \
  -D Tpetra_INST_PTHREAD:BOOL=${PTHREAD} \
  -D Tpetra_INST_CUDA:BOOL=${CUDA} \
  -D Tpetra_ENABLE_TESTS:BOOL=${TESTS} \
  -D Tpetra_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
-D Trilinos_ENABLE_TrilinosCouplings=ON \
  -D TrilinosCouplings_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
-D Trilinos_ENABLE_Triutils=ON \
-D Trilinos_ENABLE_Xpetra=ON \
  -D Xpetra_ENABLE_TESTS:BOOL=${TESTS} \
  -D Xpetra_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
  -D Xpetra_ENABLE_Experimental:BOOL=ON \
  -D Xpetra_ENABLE_Kokkos_Refactor:BOOL=ON \
-D Trilinos_ENABLE_Zoltan=ON \
-D Trilinos_ENABLE_Zoltan2=ON \
  -D Zoltan2_ENABLE_TESTS:BOOL=${TESTS} \
  -D Zoltan2_ENABLE_EXAMPLES:BOOL=${EXAMPLES} \
\
  -D Trilinos_ENABLE_STKMesh:BOOL=ON \
  -D Trilinos_ENABLE_STKSimd:BOOL=ON \
  -D Trilinos_ENABLE_STKIO:BOOL=ON \
  -D Trilinos_ENABLE_STKTransfer:BOOL=ON \
  -D Trilinos_ENABLE_STKSearch:BOOL=ON \
  -D Trilinos_ENABLE_STKUtil:BOOL=ON \
  -D Trilinos_ENABLE_STKTopology:BOOL=ON \
  -D Trilinos_ENABLE_STKClassic:BOOL=OFF \
  -D Trilinos_ENABLE_SEACASExodus:BOOL=ON \
  -D Trilinos_ENABLE_SEACASEpu:BOOL=ON \
  -D Trilinos_ENABLE_SEACASExodiff:BOOL=ON \
  -D Trilinos_ENABLE_SEACASNemspread:BOOL=ON \
  -D Trilinos_ENABLE_SEACASNemslice:BOOL=ON \
  -D Trilinos_ENABLE_SEACASAprepro_lib:BOOL=ON \
  -D Kokkos_ENABLE_DEPRECATION_WARNINGS=OFF \
 -DKokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
  -D Stratimikos_test_single_belos_thyra_solver_driver_nos1_nrhs8_MPI_1_DISABLE=ON \
  -D ROL_adapters_tpetra_test_sol_TpetraSimulatedConstraintInterfaceCVaR_MPI_4_DISABLE=ON \
  -D ROL_test_algorithm_TypeB_ProjectedGradient_MPI_1_DISABLE=ON \
  -D MueLu_MatrixMatrixKernelDriver_MPI_1_DISABLE=ON \
  -D Xpetra_MultiVector_UnitTests_MPI_4_DISABLE=ON \
$TRILINOS_DIR
```

--- a/core/src/Kokkos_CopyViews.hpp
+++ b/core/src/Kokkos_CopyViews.hpp
@@ -3223,9 +3223,8 @@ struct MirrorType {
 
 template <class T, class... P, class... I>
 inline typename std::enable_if<
-    std::is_same<typename ViewTraits<T, P...>::specialize, void>::value &&
-        !std::is_same<typename Kokkos::ViewTraits<T, P...>::array_layout,
-                      Kokkos::LayoutStride>::value,
+    !std::is_same<typename Kokkos::ViewTraits<T, P...>::array_layout,
+                  Kokkos::LayoutStride>::value,
     typename Kokkos::View<T, P...>::HostMirror>::type
 create_mirror(const Kokkos::View<T, P...>& src, const I&... arg_prop) {
   using src_type = View<T, P...>;
@@ -3246,9 +3245,8 @@ create_mirror(const Kokkos::View<T, P...>& src, const I&... arg_prop) {
 
 template <class T, class... P, class... I>
 inline typename std::enable_if<
-    std::is_same<typename ViewTraits<T, P...>::specialize, void>::value &&
-        std::is_same<typename Kokkos::ViewTraits<T, P...>::array_layout,
-                     Kokkos::LayoutStride>::value,
+    std::is_same<typename Kokkos::ViewTraits<T, P...>::array_layout,
+                 Kokkos::LayoutStride>::value,
     typename Kokkos::View<T, P...>::HostMirror>::type
 create_mirror(const Kokkos::View<T, P...>& src, const I&... arg_prop) {
   using src_type = View<T, P...>;
@@ -3281,11 +3279,8 @@ create_mirror(const Kokkos::View<T, P...>& src, const I&... arg_prop) {
 
 // Create a mirror in a new space (specialization for different space)
 template <class Space, class T, class... P, class... I>
-typename std::enable_if<
-    std::is_same<typename ViewTraits<T, P...>::specialize, void>::value,
-    typename Impl::MirrorType<Space, T, P...>::view_type>::type
-create_mirror(const Space&, const Kokkos::View<T, P...>& src,
-              const I&... arg_prop) {
+typename Impl::MirrorType<Space, T, P...>::view_type create_mirror(
+    const Space&, const Kokkos::View<T, P...>& src, const I&... arg_prop) {
   return typename Impl::MirrorType<Space, T, P...>::view_type(
       Kokkos::view_alloc(src.label(), arg_prop...), src.layout());
 }
@@ -3339,8 +3334,10 @@ create_mirror_view(const Space&, const Kokkos::View<T, P...>& src,
 }  // namespace Impl
 
 template <class T, class... P>
-typename Kokkos::View<T, P...>::HostMirror create_mirror(
-    Kokkos::View<T, P...> const& v) {
+std::enable_if_t<
+    std::is_same<typename ViewTraits<T, P...>::specialize, void>::value,
+    typename Kokkos::View<T, P...>::HostMirror>
+create_mirror(Kokkos::View<T, P...> const& v) {
   return Impl::create_mirror(v);
 }
 

